fdbserver/DataDistribution.actor.cpp

```diff
@@ -2028,6 +2272,140 @@ struct DDTeamCollection : ReferenceCounted<DDTeamCollection> {
 	}
 };
 
+ACTOR Future<Void> removeBadTeams(DDTeamCollection* self) {
+	wait(self->initialFailureReactionDelay);
+	loop {
+		while(self->zeroHealthyTeams->get() || self->processingUnhealthy->get()) {
+			wait(self->zeroHealthyTeams->onChange() || self->processingUnhealthy->onChange());
+		}
+		wait(delay(FLOW_KNOBS->PREVENT_FAST_SPIN_DELAY, TaskLowPriority)); //After the team trackers wait on the initial failure reaction delay, they yield. We want to make sure every tracker has had the opportunity to send their relocations to the queue.
+		if(!self->zeroHealthyTeams->get() && !self->processingUnhealthy->get()) {
+			break;
+		}
+	}
+	wait(self->addSubsetComplete.getFuture());
+	TraceEvent("DDRemovingBadTeams", self->masterId).detail("Primary", self->primary);
+	for(auto it : self->badTeams) {
+		it->tracker.cancel();
+	}
+	self->badTeams.clear();
+	return Void();
+}
+
+ACTOR Future<Void> teamRemover(DDTeamCollection* self) {
+	state int numMachineTeamRemoved = 0;
+	loop {
+		// Wait on processingUnhealthy as removeBadTeams() does
+		loop {
+			while(self->zeroHealthyTeams->get() || self->processingUnhealthy->get()) {
+				wait(self->zeroHealthyTeams->onChange() || self->processingUnhealthy->onChange());
+			}
+			wait(delay(FLOW_KNOBS->PREVENT_FAST_SPIN_DELAY, TaskLowPriority)); //After the team trackers wait on the initial failure reaction delay, they yield. We want to make sure every tracker has had the opportunity to send their relocations to the queue.
+			if(!self->zeroHealthyTeams->get() && !self->processingUnhealthy->get()) {
+				break;
+			}
+		}
+
```

Comment:
After the loop(), should wait for the badTeamRemover(), to avoid the race between adding the bad team and remove bad team.